[t1822] fixed module/plugin loading

secretrobotron opened the pull request:
1. Removed erroneous onModuleReady call in src/editor/module
2. Cleaned up modules loader a tiny bit
3. Removed unnecessary code from src/plugin/*
4. Wrote simple tests for plugin module

secretrobotron committed:
[t1822] fixed module/plugin loading

1. Removed erroneous onModuleReady call in src/editor/module
2. Cleaned up modules loader a tiny bit
3. Removed unnecessary code from src/plugin/*
4. Wrote simple tests for plugin module

secretrobotron commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
  * If a copy of the MIT license was not distributed with this file, you can
  * obtain one at http://www.mozillapopcorn.org/butter-license.txt */
 
-(function() {
```
Note: there is a lot of strange/duplicated removed/added code in this file because I fixed its indentation and styling.

mjschranz commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/a02a2254aa2bdc5/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/a02a2254aa2bdc5/output.txt

Total script time: 0.32 mins

mjschranz commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/c6196a4d2e5cdc4/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/c6196a4d2e5cdc4/output.txt

Total script time: 0.83 mins

Server is running on http://pr656.butyr.org/templates/test.html

mjschranz commented on test/plugins/index.html:
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Butter Test Suite [Popcorn Wrapper]</title>
```
Copy pasta gone wrong!

mjschranz commented on test/plugins/index.html:
```diff
@@ -0,0 +1,25 @@
+    <script type="text/javascript" src="../../src/butter.js"></script>
+    <script type="text/javascript" src="../../external/popcorn-js/popcorn.js"></script>
+    <script type="text/javascript" src="../../external/popcorn-js/modules/player/popcorn.player.js"></script>
+    <script type="text/javascript" src="../../external/popcorn-js/plugins/text/popcorn.text.js"></script>
```
You don't really need the above three scripts included do you?

mjschranz commented on src/ui/ui.js:
```diff
@@ -146,6 +124,28 @@ define( [ "core/eventmanager", "./toggler", "./logo-spinner", "./context-button"
+    this.loadIcons = function( icons, resourcesDir ){
+      var icon, img, div;
+
+      for( icon in icons ){
```
space before opening brace

space after <code>for</code>

mjschranz commented on src/ui/ui.js:
```diff
@@ -146,6 +124,28 @@ define( [ "core/eventmanager", "./toggler", "./logo-spinner", "./context-button"
+      var icon, img, div;
+
+      for( icon in icons ){
+        if( icons.hasOwnProperty( icon ) ){
```
Same as above.
space after <code>if</code>

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+
+    DragNDrop.droppable( _container, {
+      drop: function( element ){
+        if( element.getAttribute( "data-butter-draggable-type" ) === "plugin" ){
```
space after <code>if</code> and before opening brace

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+        if( element.getAttribute( "data-butter-draggable-type" ) === "plugin" ){
+          var pluginType = element.getAttribute( "data-popcorn-plugin-type" ),
+              plugin = _this.get( pluginType );
+          if( plugin ){
```
same as above

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+          var pluginType = element.getAttribute( "data-popcorn-plugin-type" ),
+              plugin = _this.get( pluginType );
+          if( plugin ){
+            for( var i=0; i<_listContainer.childNodes.length; ++i ){
```
same as above

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+              plugin = _this.get( pluginType );
+          if( plugin ){
+            for( var i=0; i<_listContainer.childNodes.length; ++i ){
+              if( _listContainer.childNodes[ i ].getAttribute( "data-popcorn-plugin-type" ) === pluginType ){
```
same as above

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+
+          var head = document.getElementsByTagName( "HEAD" )[ 0 ];
+          for ( i = 0, l = head.children.length; i < l; i++ ) {
+            if( head.children[ i ].getAttribute( "src" ) === plugin.path ) {
```
space after <code>if</code>

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+          var tracks = butter.tracks;
+          for ( i = 0, l = tracks.length; i < l; i++ ) {
+            var trackEvents = tracks[ i ].trackEvents;
+            for( var k = 0, ln = trackEvents.length - 1; ln >= k; ln-- ) {
```
space after <code>for</code>

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+      var i, l;
+
+      for ( i = 0, l = _plugins.length; i < l; i++ ) {
+        if( _plugins[ i ].name === plugin.name ) {
```
space after <code>if</code>

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
-          if( !plugin ) {
-            return;
-          }
+      if( typeof plugin === "string" ) {
```
space after <code>if</code>

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+      butter.loader.load( pluginLoadDescriptors, function(){
+        for( i = 0, l = newPlugins.length; i < l; i++ ) {
+          plugin = newPlugins[ i ];
           if( moduleOptions.defaults && moduleOptions.defaults.indexOf( plugin.type ) > -1 ){
```
You didn't technically add this code, but I'd be happy if you fixed the spacing here too.

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+        plugins = [ plugins ];
+      }
+
+      for( i = 0, l = plugins.length; i < l; i++ ) {
```
space after <code>for</code>

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+    _container.appendChild( _scrollbar.element );
+
+    this._start = function( onModuleReady ){
+      if( butter.ui ){
```
space after <code>if</code> and before opening brace

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+    }
+
+    // dial back blue/greens a bit
+    if( hue > 160 && hue < 200 ){
```
space after <code>if</code> and before opening brace

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+    lightness = ( ( srcString.charCodeAt( 1 ) + srcString.charCodeAt( 3 ) + srcString.charCodeAt( 5 ) + srcString.charCodeAt( 7 ) ) % 100 ) * 0.20 + 40;
+
+    // bump up reds because they're hard to see
+    if( hue < 20 || hue > 340 ){
```
space after <code>if</code> and before opening brace

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+    var hue = 0, saturation = 0, lightness = 0, srcString = type;
+
+    // very simple hashing function
+    while( srcString.length < 9 ){
```
space after <code>while</code> and before opening brace

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+        butter.ui.areas.tools.addComponent( _container );
+        PluginList( butter );
+      }
+      if( moduleOptions && moduleOptions.plugins ){
```
same as above

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+    var _scrollbar = new Scrollbars.Vertical( _listWrapper, _listContainer );
+    _container.appendChild( _scrollbar.element );
+
+    this._start = function( onModuleReady ){
```
space before opening brace

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+          for ( i = 0, l = tracks.length; i < l; i++ ) {
+            var trackEvents = tracks[ i ].trackEvents;
+            for( var k = 0, ln = trackEvents.length - 1; ln >= k; ln-- ) {
+              if( trackEvents[ ln ].type === plugin.name ) {
```
space after <code>if</code>

mjschranz commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+      _scrollbar.update();
+    };
+
+    this.clear = function () {
```
why the space after <code>function</code> here?

mjschranz commented on src/ui/ui.js:
```diff
@@ -146,6 +124,28 @@ define( [ "core/eventmanager", "./toggler", "./logo-spinner", "./context-button"
       });
     }
 
+    this.loadIcons = function( icons, resourcesDir ){
```
space before <code>{</code>

mjschranz commented on test/plugins/plugins.unit.js:
```diff
@@ -0,0 +1,69 @@
+/*global Butter,asyncTest,equal,start,ok*/
+require( [], function() {
```
Why have require here if you aren't loading anything?

I'm really curious. Trying to figure out if similar issues in the other tests I wrote are what's causing some problems on test swarm

secretrobotron replied:
They might be. I wasn't sure what you were doing here, but if they're not required (no pun intended), I'll remove them here.

mjschranz commented on test/plugins/plugins.unit.js:
```diff
@@ -0,0 +1,69 @@
+  // should use this lifecycle, and call rememberButter() for all
+  // created butter instances.  Any created using createButter()
+  // already have it done automatically.
+  var butterLifeCycle = (function(){
```
Again, another general curiosity. Is this something needed for such a small amount of tests? If so, I need to get this in for mine as well.

secretrobotron replied:
I'm just preparing for having more tests, really. Seems like great boilerplate code to have in all test modules.

mjschranz commented:
/botio try

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_try** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/25038ec4845dce6/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/25038ec4845dce6/output.txt

Total script time: 1.33 mins

Your job has been submitted to the swarm! see http://norway.proximity.on.ca/user/butter

secretrobotron committed:
[t1822] reivew changes - mostly style

secretrobotron committed:
[t1822] removed `require` from plugins tests

mjschranz commented:
/botio cleanup

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_cleanup** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/d161b851bff7b72/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/d161b851bff7b72/output.txt

Total script time: 0.22 mins

mjschranz commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/88b4c4b70945401/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/88b4c4b70945401/output.txt

Total script time: 0.86 mins

Server is running on http://pr656.butyr.org/templates/test.html

dseif commented on src/plugin/module.js:
```diff
@@ -2,229 +2,218 @@
+        if ( _plugins[ i ].name === plugin.name ) {
+          var tracks = butter.tracks;
+          for ( i = 0, l = tracks.length; i < l; i++ ) {
+            var trackEvents = tracks[ i ].trackEvents;
```
Can you declare trackEvents and tracks on line 148 instead of inside the for loop

secretrobotron committed:
[t1822] review fixes - variable declarations

secretrobotron committed:
[t1822] review changes

1. Rewrote most of butter.plugin.remove (less strict)
2. Added test for butter.plugin.remove

secretrobotron committed:
[t1822] Added documentation

dseif commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/0f275b251ea92de/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/0f275b251ea92de/output.txt

Total script time: 0.31 mins

dseif commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/afbb5f0454cd907/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/afbb5f0454cd907/output.txt

Total script time: 0.90 mins

Server is running on http://pr656.butyr.org/templates/test.html
